carbon-design-system/carbon
# feat: implement codemod ibm-products-update-http-errors

Closes [6036](https://github.com/carbon-design-system/ibm-products/issues/6306)

This PR updates the `@carbon/upgrade` package with a new transform `ibm-products-update-http-errors`, to update the `HttpError403`, `HttpError404`, `HttpErrorOther` to `FullPageError`.

#### Changelog

**New**

- New transform `packages/upgrade/transforms/ibm-products-update-http-errors.js`
- Includ fixtures in the `packages/upgrade/transforms/__testfixtures__/ibm-products-update-http-errors.input.js` and `packages/upgrade/transforms/__testfixtures__/ibm-products-update-http-errors.output.js` files
- Include tests `packages/upgrade/transforms/__tests__/ibm-products-update-http-errors-test.js`
- Implement a sample `packages/upgrade/fixtures/ibm-products-update-http-errors-sample.jsx` to test the Codemod locally

**Changed**

- Update `packages/upgrade/src/upgrades.js` with a new migration entry `ibm-products-update-http-errors`

#### Testing / Reviewing

- Try out the Codemod locally by:
   - Pulling down this PR
   - Run `cd packages/upgrade`
   - Run `yarn build` to build the CLI
   - Run `./bin/carbon-upgrade.js migrate list` and verify a new entry `ibm-products-update-http-errors (Rewrites HttpError403, HttpError404, HttpErrorOther to FullPageError)` present
   - Run `npx jest transforms/__tests__/ibm-products-update-http-errors-test.js` and verify test result
   - Run `./bin/carbon-upgrade.js migrate ibm-products-update-http-errors fixtures/ibm-products-update-http-errors-sample.jsx -w` to try the Codemod. (Can select the `/carbon/packages/upgrade` as workspace)
